### PR TITLE
Filtrar usuários em raio de 10Km

### PR DESCRIPTION
**Pull request by DanilloGS**

### CODE REVIEW

File: src/Ui/src/context/userContext.tsx [hunk 1]
```diff
@@ -1,21 +1,20 @@
-import { createContext, useState, useEffect } from "react"
+import { createContext, useState } from "react"
 import { Alert } from "react-native"
-import { Position, User, UserUseCase } from "@domain/entities"
+import { ListUserParams, Position, User, UserUseCase } from "@domain/entities"
 import { GitRepository, GitUser } from "@infrastructure/dto"
 import { HttpRepository } from "@domain/repositories"
-
+import { geohashForLocation } from "geofire-common";
```
Comment (GabrielDVpereira) [inline src/Ui/src/context/userContext.tsx]: Hey, maybe it's better to create a helper to this function and inject it into the context, this way it's going to be easier to mock when testing

File: src/Ui/src/context/userContext.tsx [hunk 2]
```diff
@@ -81,8 +71,15 @@ export function UserContextProvider({
         }
     }
 
+    const listUsers = async (listUserParams: ListUserParams) => {
+        console.log('adfsfadsf');
```
Comment (GabrielDVpereira) [inline src/Ui/src/context/userContext.tsx]: debug log

File: src/Ui/src/hooks/useGeohash.ts [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+import { useEffect, useState } from "react";
+
+const RADIUS_IN_METERS = 10000
+
+type useGeohashType = {
+  position?: Position,
+  callback: (listUserParams: ListUserParams) => Promise<User[]>
+}
+
+export const useGeohash = ({position, callback}: useGeohashType) => {
```
Comment (GabrielDVpereira) [inline src/Ui/src/hooks/useGeohash.ts]: Hey, is not really clear what's the responsibility of this function. Is it the one that is getting the user in a radius of 10km?

File: src/Ui/src/context/userContext.tsx (hunk 1 above)
Comment (DanilloGS) [reply src/Ui/src/context/userContext.tsx]: Criei um helper e agora estou passando como prop para dentro do Provider

File: src/Ui/src/context/userContext.tsx (hunk 2 above)
Comment (DanilloGS) [reply src/Ui/src/context/userContext.tsx]: Feito

File: src/Ui/src/hooks/useGeohash.ts (hunk 3 above)
Comment (DanilloGS) [reply src/Ui/src/hooks/useGeohash.ts]: Renomei o hook para useFilterUsers

**Review by GabrielDVpereira — APPROVED** [review]
🚀

**Pull request merged**